etc/aysnc_events.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<!--
+  ~ @author Aligent Consulting Team
+  ~ @copyright Copyright (c) 2022 Aligent Consulting. (http://www.aligent.com.au)
+  -->
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Aligent_AsyncEvents:etc/async_events.xsd">
+    <async_event name="sales.order.saved">
+        <service class="Magento\Sales\Api\OrderRepositoryInterface" method="get"/>
+    </async_event>
+    <async_event name="sales.shipment.saved">
+        <service class="Magento\Sales\Api\ShipmentRepositoryInterface" method="get"/>
+    </async_event>
+    <async_event name="customer.saved">
+        <service class="Magento\Customer\Api\CustomerRepositoryInterface" method="get"/>
```

Comment:
Will actually have the same problem with the order repository... I'll have a think about this

Reply:
... and the shipment repository

Reply:
The order one (and maybe shipment one) is fine because it uses a regular array.

The customer one uses a `Registry` class and since it's shared, the DI just injects the same instance even when you do `objectManager->create(CustomerRepositoryInterface)`.